server/services/v1/auth.go

```diff
@@ -116,3 +283,91 @@ func (h *authService) RegisterGRPC(grpc *grpc.Server) error {
+	}
+	return token.Sub, nil
+}
+
+func getAccessTokenUsingRefreshToken(req *api.GetAccessTokenRequest, a *Auth0) (*api.GetAccessTokenResponse, error) {
+	data := url.Values{
+		"refresh_token": {req.RefreshToken},
+		"client_id":     {a.authConfig.ClientId},
+		"grant_type":    {refreshToken},
+		"scope":         {scope},
+	}
+	resp, err := http.PostForm(a.authConfig.ExternalTokenURL, data)
+	if err != nil {
+		return nil, api.Errorf(api.Code_INTERNAL, "Failed to get access token: reason = %s", err.Error())
+	}
+	defer resp.Body.Close()
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, api.Errorf(api.Code_INTERNAL, "Failed to get access token: reason = %s", err.Error())
+	}
+	bodyStr := string(body)
+	if resp.StatusCode == 200 {
+		getAccessTokenResponse := api.GetAccessTokenResponse{}
+
+		err = json.Unmarshal([]byte(bodyStr), &getAccessTokenResponse)
+		if err != nil {
+			return nil, api.Errorf(api.Code_INTERNAL, "Failed to parse external response: reason = %s", err.Error())
+		}
+		return &getAccessTokenResponse, nil
+	}
+	log.Error().Msgf("Auth0 response status code=%d", resp.StatusCode)
+	return nil, api.Errorf(api.Code_INTERNAL, "Failed to get access token: reason = %s", bodyStr)
+}
+
+func getAccessTokenUsingClientCredentials(req *api.GetAccessTokenRequest, a *Auth0) (*api.GetAccessTokenResponse, error) {
+	payload := map[string]string{}
+	payload["client_id"] = req.ClientId
+	payload["client_secret"] = req.ClientSecret
+	payload["audience"] = a.authConfig.Audience
+	payload["grant_type"] = clientCredentials
+	jsonPayload, err := json.Marshal(payload)
+	if err != nil {
+		return nil, api.Errorf(api.Code_INTERNAL, "Failed to get access token: reason = %s", err.Error())
+	}
+
+	resp, err := http.Post(a.authConfig.ExternalTokenURL, "application/json", bytes.NewBuffer(jsonPayload))
+	if err != nil {
+		return nil, api.Errorf(api.Code_INTERNAL, "Failed to get access token: reason = %s", err.Error())
+	}
+	defer resp.Body.Close()
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, api.Errorf(api.Code_INTERNAL, "Failed to get access token: reason = %s", err.Error())
+	}
+	bodyStr := string(body)
+	if resp.StatusCode == 200 {
+		getAccessTokenResponse := api.GetAccessTokenResponse{}
+
+		err = json.Unmarshal([]byte(bodyStr), &getAccessTokenResponse)
+		if err != nil {
+			return nil, api.Errorf(api.Code_INTERNAL, "Failed to parse external response: reason = %s", err.Error())
+		}
+		return &getAccessTokenResponse, nil
+	}
+	log.Error().Msgf("Auth0 response status code=%d", resp.StatusCode)
+	return nil, api.Errorf(api.Code_INTERNAL, "Failed to get access token: reason = %s", bodyStr)
+}
+
+func deleteApplication(a *Auth0, applicationId string) error {
+	return a.management.Client.Delete(applicationId)
+}
+
+// helper to read created_at metadata from client application
+func readCreatedAt(c *management.Client) int64 {
```

Comment:
Use a standard date-time format(similar to what we have in schema package) and then you can use "ToUnixNano" in the lib package.

Reply:
updated to use standard string date format and ToUnixNano.